src/cursor/cur_hs.c

```diff
@@ -1181,20 +1181,22 @@ __curhs_update(WT_CURSOR *cursor)
  *     Discard a cursor range from the history store tree.
  */
 static int
-__curhs_range_truncate(WT_CURSOR *start, WT_CURSOR *stop)
+__curhs_range_truncate(WT_TRUNCATE_INFO *trunc_info)
 {
     WT_CURSOR *start_file_cursor, *stop_file_cursor;
     WT_SESSION_IMPL *session;
 
-    session = CUR2S(start);
-    start_file_cursor = ((WT_CURSOR_HS *)start)->file_cursor;
-    stop_file_cursor = stop != NULL ? ((WT_CURSOR_HS *)stop)->file_cursor : NULL;
+    session = trunc_info->session;
+    stop_file_cursor = NULL;
+
+    start_file_cursor = ((WT_CURSOR_HS *)trunc_info->start)->file_cursor;
 
     WT_STAT_DATA_INCR(session, cursor_truncate);
 
     WT_ASSERT(session, F_ISSET(start_file_cursor, WT_CURSTD_KEY_INT));
     WT_RET(__wt_cursor_localkey(start_file_cursor));
-    if (stop != NULL) {
+    if (F_ISSET(trunc_info, WT_TRUNC_EXPLICIT_STOP)) {
+        stop_file_cursor = ((WT_CURSOR_HS *)trunc_info->stop)->file_cursor;
```

Comment:
I feel like this HS_CURSOR cast could use its own macro one day but not required for this PR.